lib/palindrome_permutation.rb

```diff
@@ -1,5 +1,46 @@
 
 def palindrome_permutation?(string)
+  return true if (string == nil || string.length == 0)
 
- 
-end # method
+  hash = {}
+
+  string.chars.each_with_index do |character|
+    hash[character] = string.chars.count(character)
+  end
```

Comment:
The `.chars.count` is O(n) and so this solution is O(n^2)

Consider:

```suggestion
  string.chars.each_with_index do |character|
    hash[character] = 0 if hash[character].nil?
    hash[character] += 1
  end
```